devel/yarn/Portfile

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        yarnpkg yarn 0.18.0 v
+
+categories          devel
+
+platforms           darwin
+supported_archs     noarch
+
+license             BSD
+
+maintainers         andrewsforge.com:code \
+                    gmail.com:isomacrte \
+                    openmaintainer
+
+description         JavaScript dependency manager
+long_description    Created by Facebook to replace NPM, yarn boasts about its \
+                    speed, security, and reliability. Most notably, yarn \
+                    dependencies are installed deterministically, whereas \
+                    NPM's dependencies are not.
+
+homepage            https://yarnpkg.com/
+
+github.tarball_from releases
+distname            ${name}-${git.branch}
+worksrcdir          dist
+checksums           rmd160 26ecfb7134e6a400fdceff7c97f535466c934dce \
+                    Sha256 8fb1843d2a1283972ff5fead9d6c9f9002de793ecec6dfec7abec908403ecd19
+
+depends_run         bin:node:nodejs6
+
+patch {
```

Comment:
I believe that you misunderstood what @larryv wanted to tell you. Not that this should be changed from `post-patch` into `patch` (you should probably never do that), but you should create a proper patch file, perhaps call it `patch-bin-yarn.diff,` put it under `files/` and add a line
```
patchfiles patch-bin-yarn.diff
```
to the Portfile.

Regular expressions are too fragile, in particular when upgrading to a newer versions and this one is definitely too complex. The `reinplace` is only used when you need to substitute `${prefix}` or so.

In fact I believe that this is what you want. I don't want to decrypt or guess what your expression tries to do, but you can replace the whole line with something similar to
```
basedir=@@PREFIX@@
```
and then do
```
post-patch {
    reinplace "s|@@PREFIX@@|${prefix}|g" ${worksrcpath}/bin/yarn
}
```
I'm not sure if that's actually the proper variable or value, but I guess it might be something in the desired direction.

(I'm sorry, I commented on the wrong place earlier, fixing this now.)